app/src/main/java/org/oppia/app/story/StoryViewModel.kt

```diff
@@ -0,0 +1,76 @@
+package org.oppia.app.story
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Transformations
+import androidx.lifecycle.ViewModel
+import org.oppia.app.fragment.FragmentScope
+import org.oppia.app.model.ChapterSummary
+import org.oppia.app.model.StorySummary
+import org.oppia.domain.topic.TopicController
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.logging.Logger
+import javax.inject.Inject
+
+@FragmentScope
+class StoryViewModel @Inject constructor(
+  private val topicController: TopicController,
+  private val logger: Logger
+) : ViewModel() {
+  private lateinit var storyId: String
+
+  val storyLiveData: LiveData<StorySummary> by lazy {
+    processStoryLiveData()
+  }
+
+  val storyNameLiveData: LiveData<String> by lazy {
+    processStoryNameLiveData()
+  }
+
+  val storyChapterLiveData: LiveData<List<ChapterSummary>> by lazy {
+    processStoryChapterLiveData()
+  }
+
+  private val storyResultLiveData: LiveData<AsyncResult<StorySummary>> by lazy {
+    topicController.getStory(storyId)
+  }
+
+  fun setStoryId(storyId: String) {
+    this.storyId = storyId
+  }
+
+  private fun processStoryLiveData(): LiveData<StorySummary> {
+    return Transformations.map(storyResultLiveData, ::processStoryResult)
+  }
+
+  private fun processStoryNameLiveData(): LiveData<String> {
+    return Transformations.map(storyResultLiveData, ::processStoryName)
+  }
+
+  private fun processStoryChapterLiveData(): LiveData<List<ChapterSummary>> {
+    return Transformations.map(storyResultLiveData, ::processStoryChapterList)
+  }
+
+  private fun processStoryResult(storyResult: AsyncResult<StorySummary>): StorySummary {
+    if (storyResult.isFailure()) {
+      logger.e("StoryFragment", "Failed to retrieve Story: " + storyResult.getErrorOrNull())
```

Comment:
Prefer passing the error directly to the logger rather than concatenating it into a string:

```kotlin
logger.e("StoryFragment", "Failed to retrieve story", storyResult.getErrorOrNull())
```

(Note the ``,`` rather than ``+``).

Reply:
Used :
```kotlin
logger.e("StoryFragment", "Failed to retrieve story", storyResult.getErrorOrNull()!!)
````
because storyResult.getErrorOrNull is `Throwable?` and logger required `Throwable`